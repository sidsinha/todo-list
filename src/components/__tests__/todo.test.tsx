import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import Todo from '../todo';

const todoProps = {
  index: 0,
  id: 1, 
  name: "Go for Walk", 
  isEditing: false
}

afterEach(cleanup);

test('render 1 todo item', () => {
    const { getByTestId } = render(<Todo {...todoProps} />);
    expect(getByTestId('todo-name')).toHaveTextContent('Go for Walk');
    expect(getByTestId('todo-edit')).toHaveTextContent('EDIT');
    expect(getByTestId('todo-delete')).toHaveTextContent('DELETE');
  });

test('render 1 todo item', () => {
    const { getByTestId } = render(<Todo {...todoProps} />);
    expect(getByTestId('todo-name')).toHaveTextContent('Go for Walk');
  });

test('click on Edit', async() => {
    render(<Todo {...todoProps} />);
    const EditLink = screen.getByTestId("todo-edit");
    fireEvent.click(EditLink);
    // expect(screen.queryByText('CANCEL')).toBeInTheDocument();
  });

test('matches snapshot', () => {
    const { container } = render(<Todo {...todoProps} />)
    expect(container.firstChild).toMatchSnapshot();
  });
