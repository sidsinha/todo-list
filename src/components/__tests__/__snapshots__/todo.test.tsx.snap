// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="bg-white ma2 pa3 flex justify-between"
>
  <span
    class="b pr2"
  >
    1
    .
  </span>
  <span
    class="tl"
    data-testid="todo-name"
    style="flex: 2;"
  >
    Go for Walk
  </span>
  <span
    class="f6 pointer mr3 bg-light-green ph3"
    data-testid="todo-edit"
  >
    EDIT
  </span>
  <span
    class="f6 pointer bg-red ph3 white hover-bg-black"
    data-testid="todo-delete"
  >
    DELETE
  </span>
</div>
`;
